src/page.ts

```diff
@@ -0,0 +1,119 @@
+import { IFilterable } from './ifilterable';
+import { IReport } from './report';
+import { Visual } from './visual';
+import * as models from 'powerbi-models';
+
+export interface IPage {
+    report: IReport;
+    name: string;
+    displayName: string;
+    setActive(): void;
+}
+
+export class Page implements IPage, IFilterable {
+    report: IReport;
+    name: string;
+    // This can be undefined in cases where page is created manually
+    displayName: string;
+
+    constructor(report: IReport, name: string, displayName?: string) {
+        this.report = report;
+        this.name = name;
+        this.displayName = displayName;
+    }
+
+    /**
+     * Gets all page level filters within report
+     * 
+     * ```javascript
+     * page.getFilters()
+     *  .then(pages => { ... });
+     * ```
+     */
+    getFilters() {
+        return this.report.service.hpm.get<models.IFilter[]>(`/report/pages/${this.name}/filters`, { uid: this.report.config.uniqueId }, this.report.iframe.contentWindow)
+            .then(response => response.body,
+            response => {
+                throw response.body;
```

Comment:
Assuming these `throw` statements will be replace once we have implementations for them on the WFE.